### PR TITLE
rule(macro `root_dir`): correct macro to exactly match the `/root` dir and not other with just `/root` as a prefix

### LINKED ISSUE #1141: The root_dir macro matches /root123abc
The `root_dir` macro is currently defined as:

```
- macro: root_dir
  condition: ((fd.directory=/ or fd.name startswith /root) and fd.name contains "/")
```

Unfortunately, this matches on any `fd.name` beginning with the word "root", even if that's not a sensitive location. This is triggering a number of alerts for containers that place files in locations like `/rootDir` and similar, causing them to be misclassified as sensitive.

To reproduce, create a container with a directory named `/rootDir` or `/root123abc` and modify a file within one of those locations. Falco will report "File below / or /root opened for writing".

Only files immediately under `/` and `/root` should match this macro.

- Falco version: 0.21.0
- System info:
{
  "machine": "x86_64",
  "nodename": "[redacted]",
  "release": "4.15.0-1065-aws",
  "sysname": "Linux",
  "version": "#69-Ubuntu SMP Thu Mar 26 02:17:29 UTC 2020"
}
- Cloud provider or hardware configuration: AWS
- OS:
```
NAME="Ubuntu"
VERSION="18.04.4 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.4 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```
- Kernel: 4.15.0-1065-aws
- Installation method: DEB

### PR DESCRIPTION
**Pull request by leogr**
**What type of PR is this?**

/kind bug
/kind rule-update

**Any specific area of the project related to this PR?**

/area rules

**What this PR does / why we need it**:

:point_right: #1141

**Which issue(s) this PR fixes**:

Fixes #1141

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
rule(macro `root_dir`): correct macro to exactly match the `/root` dir and not other with just `/root` as a prefix  
```

### CODE REVIEW

File: rules/falco_rules.yaml [hunk 1]
```diff
@@ -110,7 +110,7 @@
 
 # This detects writes immediately below / or any write anywhere below /root
 - macro: root_dir
-  condition: ((fd.directory=/ or fd.name startswith /root) and fd.name contains "/")
+  condition: ((fd.directory=/ or fd.name startswith /root/) and fd.name contains "/")
```
Comment (Kaizhe) [inline rules/falco_rules.yaml:113]: can you remove `and fd.name contains "/"` -- this part is redundant to me.

Comment (leogr) [reply rules/falco_rules.yaml]: I did not notice that! Thank you!

Fixed by https://github.com/falcosecurity/falco/pull/1279/commits/5f0d1b832a0a6b689050a96938b3b9ef8b24cadb

**Comment by leogr** [comment]
Requested change added by https://github.com/falcosecurity/falco/pull/1279/commits/5f0d1b832a0a6b689050a96938b3b9ef8b24cadb

/hold cancel

Pls @Kaizhe review this again. Thanks in advance

**Comment by leogr** [comment]
Change requested applied by https://github.com/falcosecurity/falco/pull/1279/commits/5f0d1b832a0a6b689050a96938b3b9ef8b24cadb

Please review again

/hold cancel

Thanks!

**Pull request merged**